lib/simctl.js

```diff
@@ -326,29 +331,44 @@ async function shutdown (udid) {
   }
 }
 
+/**
+ * @typedef {Object} SimCreationOpts
+ * @property {string} platform [iOS] - Platform name in order to specify runtime such as 'iOS', 'tvOS', 'watchOS'
+ * @property {number} timeout [10000] - The maximum number of milliseconds to wait
+ *                                      unit device creation is completed.
+ */
 /**
  * Create Simulator device with given name for the particular
  * platform type and version.
  *
  * @param {string} name - The device name to be created.
  * @param {string} deviceTypeId - Device type, for example 'iPhone 6'.
  * @param {string} runtimeId - Platform version, for example '10.3'.
- * @param {number} timeout [10000] - The maximum number of milliseconds to wait
- *                                   unit device creation is completed.
+ * @param {?SimCreationOpts} opts - Simulator options for creating devives
  * @return {string} The UDID of the newly created device.
  * @throws {Error} If the corresponding simctl subcommand command
  *                 returns non-zero return code.
  */
-async function createDevice (name, deviceTypeId, runtimeId, timeout = 10000) {
+async function createDevice (name, deviceTypeId, runtimeId, opts = {}) {
+  const {
+    platform = SIM_RUNTIME_NAME_SUFFIX_IOS,
+    timeout = DEFAULT_CREATE_SIMULATOR_TIMEOUT
+  } = opts;
+
   let udid;
   // first make sure that the runtime id is the right one
   // in some versions of Xcode it will be a patch version
   try {
-    runtimeId = await getRuntimeForPlatformVersion(runtimeId);
+    runtimeId = await getRuntimeForPlatformVersion(runtimeId, platform);
   } catch (err) {
     log.warn(`Unable to find runtime for iOS '${runtimeId}'. Continuing`);
   }
 
+  const cltVersion = await getCommandLineToolsVersion();
+  if (util.compareVersions(cltVersion, '>=', CMDLINE_TOOLS_FORMAT_CHANGED_VERSION)) {
+    runtimeId = `${SIM_RUNTIME_NAME}${platform}-${runtimeId.replace('.', '-')}`;
```

Comment:
should it only replace the dot once or it should be `/\./g`?

Reply:
ah... nice. haven't seen multiple dots, but it should be `-` following the rule